neural_compressor/torch/algorithms/weight_only/gptq.py

```diff
@@ -1047,7 +1047,10 @@ def fasterquant(self, W, blocksize=128, percdamp=0.01, groupsize=-1, act_order=F
         Q = torch.zeros_like(W)
 
         damp = percdamp * torch.mean(torch.diag(H))
-        diag = torch.arange(self.columns, device=self.device)
+        if "hpu" in self.device:

```

Comment:
shall we add TODO item here for tracking?

Reply:
Sure, I will.